From b3458a6243c8de5afb461d221c3992ba71d21681 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Thu, 7 Feb 2019 12:53:29 +0100
Subject: [PATCH] SNESTEST is removed

The symbol is removed in 3.10.3, which prevents compilation of dolfin,
but the functionality was removed in 3.9
---
 dolfin/nls/PETScSNESSolver.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dolfin/nls/PETScSNESSolver.cpp b/dolfin/nls/PETScSNESSolver.cpp
index b3e1d62c5..09238bed2 100644
--- a/dolfin/nls/PETScSNESSolver.cpp
+++ b/dolfin/nls/PETScSNESSolver.cpp
@@ -48,7 +48,11 @@ PETScSNESSolver::_methods
 = { {"default",      {"default SNES method", ""}},
     {"newtonls",     {"Line search method", SNESNEWTONLS}},
     {"newtontr",     {"Trust region method", SNESNEWTONTR}},
+#if PETSC_VERSION_LT(3,9,0)
+    // SNESTEST functionality removed in petsc 3.9,
+    // symbol removed in 3.10.3
     {"test",         {"Tool to verify Jacobian approximation", SNESTEST}},
+#endif
     {"ngmres",       {"Nonlinear generalised minimum residual method",
                       SNESNGMRES}},
     {"nrichardson",  {"Richardson nonlinear method (Picard iteration)",
-- 
2.19.1

